client/asset/dcr/dcr.go

```diff
@@ -2598,7 +2543,7 @@ func (dcr *ExchangeWallet) signTxAndAddChange(baseTx *wire.MsgTx, feeRate uint64
 	// Sign the transaction to get an initial size estimate and calculate
 	// whether a change output would be dust.
 	sigCycles := 1
-	msgTx, err := dcr.signTx(baseTx)
+	msgTx, err := dcr.wallet.SignRawTransaction(dcr.ctx, baseTx)
```

Comment:
Tried this before, the `baseTx` ends up getting mutated and affects subsequent calls to srt from the loop below. Since the rpc only needs an hex string, can pass that around and have other implementations do with that as needed.